elastic_service/elastic_handler.go

```diff
@@ -0,0 +1,165 @@
+package elastic_service
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/daniyakubov/book_service_n/config"
+	"github.com/daniyakubov/book_service_n/consts"
+	"github.com/daniyakubov/book_service_n/db_service"
+	"github.com/daniyakubov/book_service_n/models"
+	errors "github.com/fiverr/go_errors"
+	"github.com/olivere/elastic/v7"
+	"strconv"
+	"strings"
+)
+
+var _ db_service.DBHandler = &ElasticHandler{}
+
+type ElasticHandler struct {
+	Client *elastic.Client
+}
+
+func NewElasticHandler(client *elastic.Client) ElasticHandler {
+	return ElasticHandler{
+		client,
+	}
+}
+
+func (e *ElasticHandler) UpdateBook(ctx context.Context, title string, id string) (err error) {
+	_, err = e.Client.Update().
+		Index(consts.Index).
+		Id(id).
+		Doc(map[string]interface{}{consts.Title: title}).
+		Do(ctx)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("couldn't update book with title %s and id %s", title, id))
+	}
+	return nil
+}
+
+func (e *ElasticHandler) AddBook(ctx context.Context, body []byte) (string, error) {
```

Comment:
Perhaps using `Insert` is more correct here, WDYT?
```suggestion
func (e *ElasticHandler) InsertBook(ctx context.Context, body []byte) (string, error) {
```

Reply:
Both sounds good to me, changed it to InsertBook :)